envoy/api/os_sys_calls.h

```diff
@@ -17,6 +19,20 @@ struct EnvoyTcpInfo {
   std::chrono::microseconds tcpi_rtt;
 };
 
+// Small struct to avoid exposing ifaddrs -- which is not defined in all platforms -- to the
+// codebase.
+struct InterfaceAddress {
```

Comment:
happy to move this to address.h if we think it makes more sense there. I couldn't decide.